src/Server.php

```diff
@@ -227,17 +236,26 @@ public function handleRequest(ConnectionInterface $conn, ServerRequestInterface
             $conn->write("HTTP/1.1 100 Continue\r\n\r\n");
         }
 
-        $callback = $this->callback;
         $cancel = null;
-        $promise = new Promise(function ($resolve, $reject) use ($callback, $request, &$cancel) {
-            $cancel = $callback($request);
+        $promise = new Promise\Promise(function ($resolve, $reject) use ($request, &$cancel) {
+            $callable = $this->callable;
```

Comment:
`$this` isn't available in PHP 5.3, see https://travis-ci.org/reactphp/http/jobs/271815811